sql/src/main/java/io/crate/planner/PlanNodeBuilder.java

```diff
@@ -160,6 +203,68 @@ public static CollectNode collect(TableInfo tableInfo,
                                List<Symbol> toCollect,
                                ImmutableList<Projection> projections,
                                @Nullable String partitionIdent) {
-        return collect(tableInfo, whereClause, toCollect, projections, partitionIdent, null);
+        return collect(tableInfo, whereClause, toCollect, projections, partitionIdent, null, null, null);
+    }
+
+    public static GroupByCollectNodeInfo createGroupByCollectNodeInfo(QuerySpec querySpec, SplitPoints splitPoints) {
+        OrderBy orderBy = null;
+        Integer limit = null;
+        if (!querySpec.hasAggregates()) {
+            orderBy = querySpec.orderBy();
+            if(orderBy == null) {
+                // No orderBy given, order by groupKeys
+                orderBy = OrderBy.fromSymbols(querySpec.groupBy());
+            }
+            orderBy = orderBy != null && orderBy.hasAggregation() ? null : orderBy;
```

Comment:
if orderBy was null then hasAggregation MUST be falls as group by keys cannot contain aggregations. So the logic here can probably be simplified